src/utils.ts

```diff
@@ -0,0 +1,125 @@
+import { publish as _publish } from 'libnpmpublish'
+import got from 'got'
+import ssri from 'ssri'
+import URL from "url"
+import {MigrationSettings} from "./settings";
+
+const logger = require('pino')()
+
+export function getScopedOptions(packageName: string, source: Record<string, string>) {
+  const scope = getNamedScope(packageName)
+  return scopedOptions(scope, source.registry, source.token)
+}
+
+// exposed for testing
+export function getNamedScope(name: string) {
+  const scope = name.match(/^(@[\w-]+)\/[\w-]+$/)
+  return scope ? scope[1] : undefined
+}
+
+function scopedOptions(scope: string | undefined, registry: string, token: string | undefined) {
+  const scopedRegistry = scope ? scope + ':registry' : 'registry'
+  const opts: Record<string, any> = {}
+  opts[scopedRegistry] = registry
+  if (token) {
+    opts.token = token
+  }
+  return opts
+}
+
+export interface ManifestDist {
+  integrity: string
+  shasum: string
+  tarball: string
```

Comment:
Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/joebowbeer/regsync/pullRequest?prid=9263162)
```suggestion
  tarball: string;
```